inappmessaging/USERGUIDE.md

```diff
@@ -120,16 +120,25 @@ class AppUserInfoProvider : UserInfoProvider {
 * **The internal IAM backend only supports production RAE token.**
 
 ### #6 Initializing In-App Messaging SDK.
-In-App Messaging is automatically initialized at startup and handled on a background thread.
-Host app just needs to register the provider containing the user information.
+Host app should initialize the SDK, then register the provider containing the user information.
 
 In your Application class' `onCreate()` method, add:
 
 ```kotlin
-InAppMessaging.instance().registerPreference(YourUserInfoProvider())
+// API: init(context: Context, errorCallback: ((ex: Exception) -> Unit)? = null): Boolean
+// `errorCallback` lambda function is optional,
+// and can be used for analytics and logging of encountered initialization issues.
+val iamFlag = InAppMessaging.init(this) {
+    Log.e(TAG, it.localizedMessage, it.cause)
+}
+
+// use flag to enable/disable IAM feature in your app.
+if (iamFlag) {
+    InAppMessaging.instance().registerPreference(provider)
```

Comment:
what happens when API is called when init returned false? Or where API is called before init?

Reply:
Nothing will happen since `instance` is initialized as `NotInitializedInAppMessaging()`